src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementEventNormalizationService.cs

```diff
@@ -201,5 +189,70 @@ private Task<bool> ProcessErrorAsync(Exception ex, EventData data)
             var handled = _exceptionTelemetryProcessor.HandleException(ex, _log);
             return Task.FromResult(!handled);
         }
+
+        private IEnumerable<(string, IMeasurement)> CreateMeasurementIterator(JToken token, string partitionId, ConcurrentBag<Exception> exceptions, Func<Exception, EventData, Task<bool>> errorConsumer, EventData evt)
+        {
+            IEnumerator<Measurement> enumerable = null;
+            Action<Exception> storeNormalizedException = ex =>
+            {
+                var normalizationException = new NormalizationDataMappingException(ex);
+                if (errorConsumer(normalizationException, evt).ConfigureAwait(false).GetAwaiter().GetResult())
+                {
+                    exceptions.Add(normalizationException);
+                }
+            };
+
+            try
+            {
+                enumerable = _contentTemplate.GetMeasurements(token).GetEnumerator();
```

Comment:
Shouldn't be we able to remove the lower while loop and just yield return as we iterate through each element here?

Reply:
See my comment about `yield` statements inside of try/catch blocks.

Reply:
Switching this to an `AsyncEnumberable` could be a good idea. At the very least it would simplify things. I did make use of the pipelines ability to batch, though. But we could see if that could be refactored.

Reply:
In the other perf PR we are no longer using this class. Thoughts on reverting this change?  It will still be used for the Azure Function implementation but we wouldn't be taking the risk of potentially introducing a new bug for that path.  The new metric is great, and we can work to port it over to the new implementation.